### PR TITLE
Try about #169 - URI getter on WSResponse

### PR DESCRIPTION
**Pull request by cchantep**

### CODE REVIEW

File: integration-tests/src/test/scala/play/libs/ws/ahc/AhcWSClientSpec.scala [hunk 1]
```diff
@@ -99,10 +99,11 @@ class AhcWSClientSpec(implicit executionEnv: ExecutionEnv) extends Specification
         val responseXml = response.getBody(xml())
         responseXml.normalizeDocument()
 
-        responseXml.isEqualNode(document) must beTrue
+        responseXml.isEqualNode(document) must beTrue and {
```
Comment (marcospereira) [inline integration-tests/src/test/scala/play/libs/ws/ahc/AhcWSClientSpec.scala]: Could you please add tests to the files below instead:

1. Java: https://github.com/playframework/play-ws/blob/master/play-ahc-ws-standalone/src/test/scala/play/libs/ws/ahc/AhcWSResponseSpec.scala
2. Scala: https://github.com/playframework/play-ws/blob/master/play-ahc-ws-standalone/src/test/scala/play/api/libs/ws/ahc/AhcWSResponseSpec.scala

They are a better place to test this new method.

Comment (cchantep) [inline integration-tests/src/test/scala/play/libs/ws/ahc/AhcWSClientSpec.scala]: I don't think so, as for other methods depending on the underlying AHC which are only tested there.

Comment (marcospereira) [inline integration-tests/src/test/scala/play/libs/ws/ahc/AhcWSClientSpec.scala]: Hi @cchantep, sorry, I don't get your point about why the files above aren't a good place to test `getUri`/`uri` methods and why do you think here is a better place.

Comment (cchantep) [inline integration-tests/src/test/scala/play/libs/ws/ahc/AhcWSClientSpec.scala]: There is quite no value to test getters depending on the mock used in the unit tests

Comment (marcospereira) [inline integration-tests/src/test/scala/play/libs/ws/ahc/AhcWSClientSpec.scala]: Sorry for not being clear.

I was not suggesting to use mocks to test this, but instead, have a real client (with a "real" server) testing the following scenarios:

1. A simple GET to a server responding HTTP status 200 should return the requested url
2. A simple GET to a server responding HTTP status 30x should return the final url when follow redirect is enabled
3. A simple GET to a server responding HTTP status 30x should return the requested url when follow redirect is disabled
4. Maybe a POST to a server responding HTTP status 307 should

WDYT?

Comment (cchantep) [inline integration-tests/src/test/scala/play/libs/ws/ahc/AhcWSClientSpec.scala]: So that would be integration tests ... and so would be there

Comment (marcospereira) [inline integration-tests/src/test/scala/play/libs/ws/ahc/AhcWSClientSpec.scala]: Yes, integration tests. Agree they should be in `integration-tests` subproject.

**Comment by marcospereira** [comment]
Hi @cchantep, this now has conflicts because of #180.

**Comment by cchantep** [comment]
Updated @marcospereira

**Comment by marcospereira** [comment]
Build is broken when running the binary compatibility check. Maybe you had some problem while rebasing?

**Comment by cchantep** [comment]
Add no issue at rebase (as indicated by the status there on GitHub), and for now cannot reproduce the MiMa error locally

**Comment by marcospereira** [comment]
Weird. Let me trigger another build to see if it was a glitch.

**Comment by cchantep** [comment]
Fixed

**Comment by cchantep** [comment]
Up

**Comment by mkurz** [comment]
Branch has conflicts.

**Comment by marcospereira** [comment]
Hi @cchantep, there are some conflicts here. Besides that, this looks good to me.

**Pull request merged**